Doras/src/app/roles/roles.component.ts

```diff
@@ -5,79 +5,94 @@ import { Band } from '../models/Band';
+  constructor(private data: DataService, private dataTransferService : DataTransferService, private responsibilities: ResponsibilitiesComponent, private switchBoard: SwitchBoardService) { }
 
   ngOnInit() {
     this.data.getBand().subscribe(c => {
       this.bands = c.reverse();
 
-      let newArray = [];
-      let chunkOfArray = 3;
-      let nextValue;
-      let length = this.bands.length;
-      for (let i = 0; i < length; i+=chunkOfArray) {
-        nextValue = i + 3;
-        newArray.push(this.bands.slice(i, nextValue));
+      let tempArray: any[] = [];
+      let step: number = 3;
+      let nextSplitValue: number;
+      let length: number = this.bands.length;
+      for (let i = 0; i < length; i+=step) {
+        nextSplitValue = i + 3;
+        tempArray.push(this.bands.slice(i, nextSplitValue));
       }
-      this.endArray = newArray;
+      this.jobBandArray = tempArray;
     })
   }
 
   onSlide(slideEvent: NgbSlideEvent){
      if (slideEvent.source === NgbSlideEventSource.ARROW_LEFT){
-        if (this.count === 0) {
-          this.count = 2;
+        if (this.pageCount === 0) {
+          this.pageCount = 2;
         } else {
-          this.count--;
+          this.pageCount--;
         }
     } else if ( slideEvent.source === NgbSlideEventSource.ARROW_RIGHT) {
-      if (this.count === 2) {
-        this.count = 0;
+      if (this.pageCount === 2) {
+        this.pageCount = 0;
       } else {
-        this.count++;
+        this.pageCount++;
       }
     }
 
-    this.data.getJobRoleTitle(this.capability.capId, this.endArray[this.count][0].bandId).subscribe(c => {
+    this.data.getJobRole(this.capability.capId, this.jobBandArray[this.pageCount][0].bandId).subscribe(c => {
       this.jobs1 = c;
-      let obj = {};
-      this.jobs1.forEach(item => obj[item.bandId] = item.title);
-      this.job1Title = Object.values(obj)[0];
+      if (!this.jobs1[0]) {
+        this.jobs1.push(undefined);
+      }
     })
 
-    this.data.getJobRoleTitle(this.capability.capId, this.endArray[this.count][1].bandId).subscribe(c => {
+    this.data.getJobRole(this.capability.capId, this.jobBandArray[this.pageCount][1].bandId).subscribe(c => {
       this.jobs2 = c;
-      let obj = {};
-      this.jobs2.forEach(item => obj[item.bandId] = item.title);
-      this.job2Title = Object.values(obj)[0];
+      if (!this.jobs2[0]) {
+        this.jobs2.push(undefined);
+      }
   })
-    this.data.getJobRoleTitle(this.capability.capId, this.endArray[this.count][2].bandId).subscribe(c => {
+    this.data.getJobRole(this.capability.capId, this.jobBandArray[this.pageCount][2].bandId).subscribe(c => {
       this.jobs3 = c;
-      let obj = {};
-      this.jobs3.forEach(item => obj[item.bandId] = item.title);
-      this.job3Title = Object.values(obj)[0];
+      if (!this.jobs3[0]) {
+        this.jobs3.push(undefined);
+      }
   })
 }
 
+openResponsibility1() {
+  this.switchBoard.getJob(this.jobs1);
```

Comment:
this is being repeated three times.... should be extracted into a private method which takes in a parameter array and call that method 3 times instead